src/handler.ts

```diff
@@ -17,48 +17,32 @@ const handler = async (
     `\nRunning Service:\n '${SERVICE}'\n mode: ${NODE_ENV}\n stage: '${AWS_STAGE}'\n region: '${AWS_REGION}'\n 
   Send to smc: ${SEND_TO_SMC}\n`,
   );
-  if (SEND_TO_SMC != undefined && SEND_TO_SMC.toUpperCase() === 'TRUE') {
-    try {
-      logger.debug(`Function triggered with '${JSON.stringify(event)}'.`);
 
-      for (const record of event.Records) {
+  const batchItemFailures: SQSBatchItemFailure[] = [];
+
+  if (SEND_TO_SMC?.toUpperCase() === 'TRUE') {
+    logger.debug(`Function triggered with '${JSON.stringify(event)}'.`);
+
+    for (const record of event.Records) {
+      try {
         const dynamoDBEvent: DynamoDBRecord = JSON.parse(record.body) as DynamoDBRecord;
-        if (dynamoDBEvent){
-          const mcRequests: MCRequest[] = extractMCTestResults(dynamoDBEvent);
+        const mcRequests: MCRequest[] = extractMCTestResults(dynamoDBEvent);
 
-          if (mcRequests != null && mcRequests.length > 0) {
-            await sendMCProhibition(mcRequests);
-          } else {
-            logger.info(`No relevant MC test results found in the record: ${JSON.stringify(dynamoDBEvent)}`);
-          }
+        if (mcRequests.length > 0) {
```

Comment:
are we sure mcRequests can't return null or undefined, means this would error needlessly

Reply:
extractMCTestResults should be returning an empty array if nothing is found. Looks like the only non-stardard return is the 400 HttpError that is thrown if there is an issue when validating the McRequest.